pkg/controller/kogitotrusty/kogitotrusty_controller.go

```diff
@@ -0,0 +1,156 @@
+	err = c.Watch(&source.Kind{Type: &appv1alpha1.KogitoTrusty{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner KogitoTrusty
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &appv1alpha1.KogitoTrusty{},
+	})
+	if err != nil {
+		return err
+	}
+
+	// We also watch for any resources regarding infra to recreate it in case is deleted and we depend on them
+	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForOwner{IsController: true, OwnerType: &appv1alpha1.KogitoInfra{}})
+	if err != nil {
+		return err
+	}
+
+	watchedObjects := []framework.WatchedObjects{
+		{
+			GroupVersion: routev1.GroupVersion,
+			AddToScheme:  routev1.Install,
+			Objects:      []runtime.Object{&routev1.Route{}},
+		},
+		{
+			GroupVersion: imgv1.GroupVersion,
+			AddToScheme:  imgv1.Install,
+			Objects:      []runtime.Object{&imgv1.ImageStream{}},
+		},
+		{Objects: []runtime.Object{&corev1.Service{}, &appsv1.Deployment{}}},
+	}
+	controllerWatcher := framework.NewControllerWatcher(r.(*ReconcileKogitoTrusty).client, mgr, c, &appv1alpha1.KogitoTrusty{})
+	if err = controllerWatcher.Watch(watchedObjects...); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileKogitoTrusty implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileKogitoTrusty{}
+
+// ReconcileKogitoTrusty reconciles a KogitoTrusty object
+type ReconcileKogitoTrusty struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client *client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a KogitoTrusty object and makes changes based on the state read
+// and what is in the KogitoTrusty.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileKogitoTrusty) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.With("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling KogitoTrusty")
+
+	reqLogger.Infof("Injecting Trusty Index URL into KogitoApps in the namespace '%s'", request.Namespace)
+	if err := infrastructure.InjectTrustyURLIntoKogitoApps(r.client, request.Namespace); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	instances := &appv1alpha1.KogitoTrustyList{}
+	definition := services.ServiceDefinition{
+		DefaultImageName:    infrastructure.DefaultTrustyImageName,
+		Request:             request,
+		KafkaTopics:         kafkaTopics,
+		RequiresPersistence: true,
```

Comment:
persistence and events are always required?

Reply:
Yes, the trusty service always needs kafka and a storage for the persistance of the tracing events